trace/trace.go

```diff
@@ -2,25 +2,44 @@ package trace
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
 	"sync"
 	"time"
 
 	"github.com/google/uuid"
 	"github.com/honeycombio/beeline-go/client"
-	"github.com/honeycombio/beeline-go/propagation"
 	"github.com/honeycombio/beeline-go/sample"
 	libhoney "github.com/honeycombio/libhoney-go"
+	otelprop "go.opentelemetry.io/otel/api/propagation"
 )
 
 var GlobalConfig Config
 
+// HTTPPropagator wraps the interface with the same name from the OpenTelemetry Go SDK.
+type HTTPPropagator interface {
+	otelprop.HTTPPropagator
```

Comment:
All the details of interfaces fulfillment are a little fuzzy but I think it's a little dangerous to embed their propagator directly. The effect is that if they add a function to the interface our implementations no longer meet our interface and so immediately fail to compile. This makes revving versions difficult. It would be better to call out the specific functions that we need our interfaces to provide and then we can lag behind their version bumps without compatibility issues.

Reply:
Good call. Replaced with a list of the methods we need to implement in our propagators.